tasks/connectors/contrast/contrast.rb

```diff
@@ -0,0 +1,425 @@
+                        "scanner_score" => ((l["vulns"].max_by { |v| v[:severity_value] })["severity_value"]).to_i, #Must be an integer
+                        "triage_state" => a["app_library_status"].nil? ? nil : map_status_to_triage_state(a["app_library_status"]),
+                        "additional_fields" => additional_fields
+                      }
+                      finding.compact!
+                    else 
+                      vuln = {
+                        "scanner_identifier" => id,
+                        "scanner_type" => SCANNER,
+                        "scanner_score" => ((l["vulns"].max_by { |v| v[:severity_value] })["severity_value"]).to_i, #Must be an integer
+                        "severity" => ((l["vulns"].max_by { |v| v[:severity_value] })["severity_value"]).to_i, #Must be an integer
+                        "created_at" => Time.at(a["first_seen"].to_i/1000).iso8601,
+                        "last_seen_at" => a["last_seen"] = 0 ? Time.at(a["first_seen"].to_i/1000).iso8601 : Time.at(a["last_seen"].to_i/1000).iso8601,
+                        "closed_at" => nil,
+                        "status" => "open",
+                        "details" => details
+                      }
+                      vuln.compact!
+                    end
+        
+                    vuln_def = {
+                      "scanner_identifier" => id,
+                      "scanner_type" => SCANNER,
+                      "cve_identifiers" => cves.join(","),
+                      "name" => "The library #{l["file_name"]} has #{l["total_vulnerabilities"]} CVEs",
+                      "description" => "#{contrast_use_https ? "https://" : "http://"}#{contrast_host}/static/ng/index.html#/#{contrast_org_id}/libraries/java/#{l["hash"]}",
+                      "solution" => solution
+                    }
+                    vuln_def.compact!
+        
+                    # Create the KDI entries
+                    create_kdi_asset(asset)
+                    if kenna_appsec_module == true
+                      create_kdi_asset_finding(asset, finding)
+                    else
+                      create_kdi_asset_vuln(asset, vuln)
+                    end
+                    create_kdi_vuln_def(vuln_def)
+                    results=true;
+                  end
+                end
+              end
+            rescue => error
+              print_error "Error processing #{l["file_name"]}: #{error.message}"
+            end  
+          end
+        end
+
+        if results == true
+          ### Write KDI format
+          kdi_output = { skip_autoclose: false, assets: @assets, vuln_defs: @vuln_defs }
+          output_dir = "#{$basedir}/#{@options[:output_directory]}"
+          filename = "generator.kdi.json"
+          write_file output_dir, filename, JSON.pretty_generate(kdi_output)
+          print_good "Output is available at: #{output_dir}/#{filename}"
```

Comment:
can we change this to: 
kdi_upload @output_dir, filename, @kenna_connector_id, @kenna_api_host, @kenna_api_key, false, 3, 1

it will write the file out with a newer more efficient streaming method and automatically upload if the kenna params are not nil
@davidaustinarcher